### ISSUE TITLE
[dev/api] Some EnvironmentTests fail in non-Windows netcoreapp1.1 runs

### ISSUE THREAD

**Issue by joperezr** [dotnet/corefx#11306]
cc: @stephentoub 

```
./corerun xunit.console.netcore.exe System.Runtime.Extensions.Tests.dll -xml testResults.xml -notrait Benchmark=true -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing -notrait category=nonosxtests
xUnit.net console test runner (64-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Runtime.Extensions.Tests
Discovered:  System.Runtime.Extensions.Tests
Starting:    System.Runtime.Extensions.Tests
   System.Tests.EnvironmentTests.GetFolderPath_Unix_NonEmptyFolderPaths(folder: CommonTemplates) [FAIL]
      Assert+WrapperXunitException : File path: /Users/joperezr/Desktop/git/corefx/src/System.Runtime.Extensions/tests/System/EnvironmentTests.cs. Line: 224
      ---- Assert.NotEmpty() Failure
      Stack Trace:
            at Assert.WrapException(Exception e, String callerFilePath, Int32 callerLineNumber)
            at Assert.NotEmpty(IEnumerable collection, String path, Int32 line)
            at System.Tests.EnvironmentTests.GetFolderPath_Unix_NonEmptyFolderPaths(SpecialFolder folder)
         ----- Inner Stack Trace -----
            at Assert.NotEmpty(IEnumerable collection, String path, Int32 line)
   System.Tests.EnvironmentTests.GetFolderPath_Unix_NonEmptyFolderPaths(folder: Templates) [FAIL]
      Assert+WrapperXunitException : File path: /Users/joperezr/Desktop/git/corefx/src/System.Runtime.Extensions/tests/System/EnvironmentTests.cs. Line: 224
      ---- Assert.NotEmpty() Failure
      Stack Trace:
            at Assert.WrapException(Exception e, String callerFilePath, Int32 callerLineNumber)
            at Assert.NotEmpty(IEnumerable collection, String path, Int32 line)
            at System.Tests.EnvironmentTests.GetFolderPath_Unix_NonEmptyFolderPaths(SpecialFolder folder)
         ----- Inner Stack Trace -----
            at Assert.NotEmpty(IEnumerable collection, String path, Int32 line)
   System.Tests.EnvironmentTests.GetFolderPath_Unix_NonEmptyFolderPaths(folder: MyVideos) [FAIL]
      Assert+WrapperXunitException : File path: /Users/joperezr/Desktop/git/corefx/src/System.Runtime.Extensions/tests/System/EnvironmentTests.cs. Line: 224
      ---- Assert.NotEmpty() Failure
      Stack Trace:
            at Assert.WrapException(Exception e, String callerFilePath, Int32 callerLineNumber)
            at Assert.NotEmpty(IEnumerable collection, String path, Int32 line)
            at System.Tests.EnvironmentTests.GetFolderPath_Unix_NonEmptyFolderPaths(SpecialFolder folder)
         ----- Inner Stack Trace -----
            at Assert.NotEmpty(IEnumerable collection, String path, Int32 line)
Finished:    System.Runtime.Extensions.Tests

=== TEST EXECUTION SUMMARY ===
   System.Runtime.Extensions.Tests  Total: 922, Errors: 0, Failed: 3, Skipped: 0, Time: 3.107s
Finished running tests. End time=17:17:34. Return value was 3
```

**Comment by stephentoub** [comment]
@joperezr, what OS?  These fail on both Linux and macOS?

**Comment by joperezr** [comment]
I saw them failing in all non-Windows for my CI, but I only ran manually on OSX. The error above is comming from my manual OSX run.

**Comment by stephentoub** [comment]
Thanks. I have a fix, but I'll wait to submit it until your change goes in so that CI will run them.

**Comment by joperezr** [comment]
Awesome! I'm expecting CI to be green now as I think I've commented out (and created issues) for all of the tests that were failing. I'll merge soon.

**Comment by stephentoub** [comment]
Fixed by #11337
